Add Russian rhyming task

Latand opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. We encourage partial PR's with ~5-10 example that we can then run the evals on and share the results with you so you know how your eval does with GPT-4 before writing all 100 examples.

## Eval details 📑
### Eval name
Russian Rhyming Patterns for kids.

### Eval description
This evaluation is designed to test GPT-4's ability to generate Russian phrases that follow a particular rhyming pattern. The system will be given a syllable and must generate a list of phrases that end with words that rhyme with that syllable. 


### What makes this a useful eval?

The eval is designed to learn Russian rhyming patterns is a valuable addition as it can improve the language generation abilities of rhymes of the model. Rhyming is an important aspect of poetry and literature, and being able to detect and generate rhyming words and phrases can make the generated text more engaging and interesting for readers. 

Right now, GPT-3.5 and GPT-4 are fairly good at English rhyming, but not in Russian or Ukrainian (they fail most of the time)

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [ ] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [ ] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [ ] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [ ] Include at least 100 high quality examples (it is okay to only contribute 5-10 meaningful examples and have us test them with GPT-4 before adding all 100)

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should
- [ ] Check that your data is in `evals/registry/data/{name}`
- [ ] Check that your yaml is registered at `evals/registry/evals/{name}.jsonl`
- [ ] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [ ] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [ ] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [ ] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [ ] I have filled out all required fields in the evals PR form
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "АЙ"}], "ideal": "АЙ-АЙ-АЙ: Говорящий попугай"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ОЙ"}], "ideal": "ОЙ-ОЙ-ОЙ: Слоненок с косой"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "АХ"}], "ideal": "АХ-АХ-АХ: Панцирь есть у черепах"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "БЫ"}], "ideal": "БЫ-БЫ-БЫ: Еж несет грибы"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ДЕ"}], "ideal": "ДЕ-ДЕ-ДЕ: Утенок плещется в воде"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ВА"}], "ideal": "ВА-ВА-ВА: К нам летит сова"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ЗА"}], "ideal": "ЗА-ЗА-ЗА: Прозрачная медуза"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ЗА"}], "ideal": "ЗА-ЗА-ЗА: Начинается гроза"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ИК"}], "ideal": "ИК-ИК-ИК: Я катаю грузовик"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ЫШЬ"}], "ideal": "ЫШЬ-ЫШЬ-ЫШЬ: Пробежала мышь"}
{"input": [{"role": "system", "content": "I want you to act as a Russian language teacher. You need to create a list of repeating syllables and phrases that will rhyme with them. An example of one line is: АБ-АБ-АБ: Вот большущий краб. The \"АБ\" part is the syllable that rhymes with the word \"краб\""}, {"role": "assistant", "content": "Certainly! I understand the task and I'll be happy to help you create rhyming phrases in Russian. Please provide the syllable you'd like me to work with."}, {"role": "user", "content": "ЩИ"}], "ideal": "ЩИ-ЩИ-ЩИ: Мы купили овощи"}
  ```
</details>

Latand commented:
Hey, yeah thanks, didn't notice that for some reason the changes from another PR (largest country) went inside this one as well.

BTW, the countries PR is already merged.

luqman-openai reviewed (changes requested):
Thanks for notifying the `country` eval is already merged. Please remove `country` eval and it's related samples from the PR.

I would recommend model graded evaluator instead of `FuzzyMatch` with the spec `closedqa` for `ru_rhyming_phrases`. There may be many possible way to answer the prompt asking for the repeating syllables and phrases that will rhyme given a word. It's better to use model graded evaluator in these type of scenarios.

We would love to review your PR again after the above mentioned concerns are addressed.

Latand commented:
> Thanks for notifying the `country` eval is already merged. Please remove `country` eval and it's related samples from the PR.
> 
> I would recommend model graded evaluator instead of `FuzzyMatch` with the spec `closedqa` for `ru_rhyming_phrases`. There may be many possible way to answer the prompt asking for the repeating syllables and phrases that will rhyme given a word. It's better to use model graded evaluator in these type of scenarios.
> 
> We would love to review your PR again after the above mentioned concerns are addressed.

Thanks! I have removed the unrelated commit from history and replaced the `FuzzyMatch` with `modelgraded`, as you suggested. Hope I did it correctly!

Latand commented:
@luqman-openai, Seems like all checks (in GitHub Actions) have passed, and the eval is correct now.

luqman-openai reviewed (approved):
Thanks for submitting the eval! I am approving this PR.

Latand commented:
> Thanks for submitting the eval! I am approving this PR.
> 
> You should see GPT-4 API access enabled in your account in the few days.

Hey! Many thanks! I have already received gpt-4 8k context and would really appreciate it if I can receive 32k context!
